Tests/Get-CEventLog.Tests.ps1

```diff
@@ -0,0 +1,121 @@
+
+#Requires -RunAsAdministrator
+#Requires -Version 5.1
+Set-StrictMode -Version 'Latest'
+
+BeforeAll {
+    Set-StrictMode -Version 'Latest'
+
+    & (Join-Path -Path $PSScriptRoot -ChildPath 'Initialize-Test.ps1' -Resolve)
+
+    $script:logName = 'Get-CEventLogs.Tests'
+
+    function GivenEventLog
+    {
+        New-CEventLog -LogName $script:logName -Source 'Get-CEventLogs.Tests'
+    }
+
+    function GivenLogEntry
+    {
+        param(
+            [String] $Message,
+            [System.Diagnostics.EventLogEntryType] $EntryType = 'Error'
+        )
+        [System.Diagnostics.EventLog]::WriteEntry($script:logName, $Message, $EntryType)
+    }
+
+    function WhenGettingEventLogs
+    {
+        param(
+            [switch] $List,
+            [String] $LogName,
+            [int] $Newest,
+            [System.Diagnostics.EventLogEntryType] $EntryType,
+            [String] $Message
+        )
+
+        $script:result = Get-CEventLog @PSBoundParameters
+    }
+
+    function ThenEventLogsAreListed
+    {
+        $script:result | Should -BeOfType 'System.Diagnostics.EventLog'
+    }
+
+    function CleanUpEventLog
+    {
+        [System.Diagnostics.EventLog]::Delete($script:logName)
+    }
+}
+
+Describe 'Get-CEventLog' {
+    BeforeEach {
+        $script:result = $null
+        Start-Sleep -Seconds 1
+    }
+
+    It 'should list event logs' {
+        WhenGettingEventLogs -List
+        ThenEventLogsAreListed
+    }
+
+    It 'should list event logs including the defaults' {
+        GivenEventLog
+        WhenGettingEventLogs -List
+        ThenEventLogsAreListed
+        $script:result.Log | Should -Contain $script:logName
+        CleanUpEventLog

```

Comment:
Instead, call `AfterEach` with a script block that does the cleanup. Also, I would probably create a `Remove-CEventLog` function that does the deletion and writes an error message if the log doesn't exist and an `Uninstall-CEventLog` that doesn't write an error if the event log doesn't exist.

```powershell
AfterEach {
    [System.Diagnostics.EventLog]::Delete($script:logName)
}
```

Reply:
This still needs to be done.